### PR TITLE
[core] Add validation for sequence group during create table

### LINKED ISSUE #2203: [Feature] Add validation for sequence group during create table
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Currently, the sequence group options is validated during the runtime. Some checks should be moved to the SchemaValidation phase.

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Aitozi**
### Purpose

Linked issue: close #2203

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java [hunk 1]
```diff
@@ -205,27 +205,8 @@ private Factory(Options options, RowType rowType) {
                     SequenceGenerator sequenceGen =
                             new SequenceGenerator(sequenceFieldName, rowType);
                     Arrays.stream(v.split(","))
-                            .map(
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java:208]: We don't need to delete here, we can keep it as it is.

**Review by JingsongLi — COMMENTED** [review]
Thanks @Aitozi left one comment

File: paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java (hunk 1 above)
Comment (Aitozi) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java]: Make sense, updated.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**